### PR TITLE
introduction of gh actions for codeql and trivy

### PR DESCRIPTION
**Pull request by ibrokethecloud**
Introduction of gh actions for codeql and trivy base scanning of repo.

### CODE REVIEW

**Review by bk201 — APPROVED** [review]
Thanks for adding the actions.

File: .github/workflows/scan.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+name: trivy-scanning
+on:
+  push:
+    branches:
+      - master
+      - 'v*'
```
Comment (guangbochen) [inline .github/workflows/scan.yml:6]:
```suggestion
      - 'v**'
```

**Review by guangbochen — APPROVED** [review]
just a small fix, everything else is LGTM

File: .github/workflows/scan.yml (hunk 1 above)
Comment (ibrokethecloud) [reply .github/workflows/scan.yml]: fixed.

**Comment by ibrokethecloud** [comment]
`golangci-lint` is coming in the next PR which fixes trivy scan issues among other items.

**Review by guangbochen — APPROVED** [review]
LGTM, thanks.

**Pull request merged**